Fixes and Improvements

Martinski4GitHub opened the pull request:
- Fixed WebUI Access Restrictions to check for valid CIDR IP address blocks.
- Fixed for changelog tag check going from NG to 3006.
- Coding improvements and fine-tuning.

Martinski4GitHub commented:
@ExtremeFiretop,

I'll be working on the issue WRT the RT-AX86S router next.
Hopefully, I'll finish it and start testing before going to bed, but I can't promise anything yet. :>).

ExtremeFiretop commented:
> @ExtremeFiretop,
> 
> I'll be working on the issue WRT the RT-AX86S router next. Hopefully, I'll finish it and start testing before going to bed, but I can't promise anything yet. :>).

I hope it's going well!

ExtremeFiretop commented:
@Martinski4GitHub 

I see lots of cleanup which is much apprechiated! I do see the changes related to Web Access Restrictions.
Just trying to find the change for the changelog now...

Question:
I noticed you didn't include any memory changes in this PR, I'm assuming that means you have more coming?

ExtremeFiretop commented:
Another bug I just noticed which is that doing a forceupdate in MerlinAU isn't updating the WebUI .asp file that is already mounted

Martinski4GitHub commented:
> FYI we have a new report from Ripshod here: https://www.snbforums.com/threads/merlinau-v1-4-6-the-ultimate-firmware-auto-updater-webui-gnuton-support.91326/post-957412
> 
> And you have a report on ntpMerlin from our user that reported a bunch of stuff but not the edge cases lol
>
>
I won't likely have time to look at the ntpMerlin report today.
Funny how he ignored my request for his "edge cases." My guess is that he couldn't find a **valid** case since I very explicitly requested to be real-world, **valid** scenarios, not just hypothetical.

ExtremeFiretop commented:
> > @Martinski4GitHub
> > I see lots of cleanup which is much apprechiated! I do see the changes related to Web Access Restrictions. Just trying to find the change for the changelog now...
> > Question: I noticed you didn't include any memory changes in this PR, I'm assuming that means you have more coming?
> 
> I was working on the modified memory management changes late last night when I noticed a couple of bugs in the latest WebGUI. I have submitted fixes in the latest PR #479. After addressing the issues, it took me some time to test and validate on MS Edge, Google Chrome, and Firefox browsers, especially the focus issue you found (great catch, BTW).
> 

Standard, I called it. I said WebUI and web access restrictions and that's exactly where you found my bugs hahaha 🤣. If I'm not confident I thought of every case, you usually come in and prove me right 😜 team work makes the dream work! Happy you found those bugs.

Wasn't hard to find the original, I happened to try to change my postpone from 7 to 15, clicked the box, typed 15, and nothing changed... Tried again and typed 15 and noticed the additional boxes on the password field increasing... I went "When did this start happening?!?!?"

> Anyway, that PR took my time away from finishing the memory changes for the RT-AX86S router. I'm almost done, but I still need to run tests and validation, first on my RT-AC86U, and then I'll ask my cousin for permission to run some tests on his RT-AX86S router, which likely won't happen until late at night after the family has gone to sleep. So just a heads-up.

No worries. I'm just happy we have some progress on that front! I'm looking forwards to seeing the suggestions

ExtremeFiretop commented:
> > FYI we have a new report from Ripshod here: https://www.snbforums.com/threads/merlinau-v1-4-6-the-ultimate-firmware-auto-updater-webui-gnuton-support.91326/post-957412
> > And you have a report on ntpMerlin from our user that reported a bunch of stuff but not the edge cases lol
> 
> I won't likely have time to look at the ntpMerlin report today. Funny how he ignored my request for his "edge cases." My guess is that he couldn't find a **valid** case since I very explicitly requested to be real-world, **valid** scenarios, not just hypothetical.

That's exactly my guess as well. When you poked for real work examples he probably came up dry. Which is fine by me. Don't worry about ntpMerlin I just wanted to laugh about that as well.

Martinski4GitHub commented:
> > > @Martinski4GitHub
> > > I see lots of cleanup which is much apprechiated! I do see the changes related to Web Access Restrictions. Just trying to find the change for the changelog now...
> > > Question: I noticed you didn't include any memory changes in this PR, I'm assuming that means you have more coming?
> > 
> > 
> > I was working on the modified memory management changes late last night when I noticed a couple of bugs in the latest WebGUI. I have submitted fixes in the latest PR #479. After addressing the issues, it took me some time to test and validate on MS Edge, Google Chrome, and Firefox browsers, especially the focus issue you found (great catch, BTW).
> 
> Standard, I called it. I said WebUI and web access restrictions and that's exactly where you found my bugs hahaha 🤣. If I'm not confident I thought of every case, you usually come in and prove me right 😜 team work makes the dream work! Happy you found those bugs.
> 
>
Yep, it's all part of a team effort.

>
> Wasn't hard to find the original, I happened to try to change my postpone from 7 to 15, clicked the box, typed 15, and nothing changed... Tried again and typed 15 and noticed the additional boxes on the password field increasing... I went "When did this start happening?!?!?"
> 
>
I very rarely use MS Edge and Chrome browsers. My go-to is Firefox, pretty much for everything, so good thing you found that bug. Again, great catch, bud!!!

Martinski4GitHub commented:
@ExtremeFiretop,
Going offline now. Got to run some errands and do grocery shopping with the wife.
I'll be back in the evening.
Just FYI.
